lib/player.js

```diff
@@ -1759,6 +1778,18 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
       this.metadataRegionTimeline_ =
           new shaka.media.RegionTimeline(() => this.seekRange());
 
+      this.metadataRegionTimeline_.addEventListener('regionadd', (event) => {
+        /** @type {shaka.extern.MetadataTimelineRegionInfo} */
+        const region = event['region'];
+        const data = new Map()
+            .set('startTime', region.startTime)
+            .set('endTime', region.endTime)
+            .set('metadataType', region.schemeIdUri)
+            .set('payload', region.payload);
+        this.dispatchEvent(shaka.Player.makeEvent_(
+            shaka.util.FakeEvent.EventName.MetadataAdded, data));
```

Comment:
I'd prefer add this code in `addMetadataToRegionTimeline_` function, to avoid have another listener